src/main/scala/io/iohk/ethereum/blockchain/sync/RegularSync.scala

```diff
@@ -46,14 +51,20 @@ trait RegularSync {
 
     case PrintStatus =>
       log.info(s"Peers: ${handshakedPeers.size} (${blacklistedPeers.size} blacklisted).")
+
+    case Done =>
+      if (waitingForActor == Option(sender())) {
+        //actor is done and we did not get response
+        scheduleResume()
+      }
```

Comment:
we are creating actor per request so it can be new actor but Done can be from previouse one.
Done is always sent when actor finishes, so we can get
Response from block headers actor
   Ask for block bodies set waiting for response (block bodies)
Done from block headers actor <- false `scheduleResume`